shot/src/main/scala/com/karumi/shot/ShotPlugin.scala

```diff
@@ -140,9 +140,11 @@ class ShotPlugin extends Plugin[Project] {
         project.getConfigurations.forEach(config => {
           facebookDependencyHasBeenAdded |= config.getAllDependencies
             .toArray(new Array[Dependency](0))
-            .exists(dependency =>
-              Config.androidDependencyGroup == dependency.getGroup
-                && Config.androidDependencyName == dependency.getName)
+            .exists(
+              dependency =>
+                Config.androidDependencyGroup == dependency.getGroup
+                  && Config.androidDependencyName == dependency.getName
+                  && Config.androidDependencyVersion == dependency.getVersion)
```

Comment:
Hum...after reviewing this issue there is something we need to review:

We can check if the version also matches but then we need to update the documentation and consider this change as a **BREAKING CHANGE**. Which will force us to update the documentation and explain the migration in the release description. Here you have the documentation change.

> If you have included in your project a dependency to related to the dexmaker and you are facing this exception: com.android.dx.util.DexException: Multiple dex files define, you can customize how the facebook SDK is added to your project and exclude the dexmaker library as follows:
```
androidTestCompile ('com.facebook.testing.screenshot:core:0.4.2') {
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker'
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker-dx'
  }
```

Should be replaced by:

```
androidTestCompile ('com.facebook.testing.screenshot:core:0.5.0') {
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker'
    exclude group: 'com.crittercism.dexmaker', module: 'dexmaker-dx'
  }
```

We should also add a note to remind the user that the version will not be added if the version name matches.

If we don't want to introduce a breaking change, we'd need to update the parser implementation and support both versions and keep this code as is and the new version will work.

The first option seems to be easier to implement, test and maintain, so I'd go for it.

Reply:
I'm including a more detailed comment in the README asking the users to use v0.5.0 of the facebook library if they want to use them both. Thanks pedro 👍

Reply:
You are welcome. Let me know when ready and I'll review the final version of this PR. @tonilopezmr would you like to publish the next release with me?

Reply:
I want to push the button! 

![](http://www.davidmcelroy.org/wp-content/uploads/2014/12/Push-the-red-button.jpg)